app/models/api/translation_memory.php

```diff
@@ -49,35 +49,23 @@
 
     /*
         We are only interested in target strings with keys in common with our
-        source strings. Not sending noise to getTranslationMemoryResults() has
-        a major performance and memory impact.
+        source strings.
     */
-    $target_strings = array_intersect_key(
-        Utils::getRepoStrings($request->parameters[4], $repository),
-        $source_strings
-    );
-
-    /*
-        We are not interested in keeping duplicate strings that have
-        different keys because this API does not take into account the
-        frequency of matches but the similarity of the strings.
-    */
-    $target_strings = array_unique($target_strings);
-
-    /*
-        The + operator is slightly faster than array_merge and also easier
-        to read. The functional difference doesn't matter in this case
-        (http://stackoverflow.com/questions/7059721/array-merge-versus/27717809#27717809)
-     */
-    $source_strings_merged += $source_strings;
-    $target_strings_merged += $target_strings;
+    $target_strings = Utils::getRepoStrings($request->parameters[4], $repository);
 
+    foreach ($source_strings as $key => $value) {
+        if (isset($target_strings[$key]) && ! empty($target_strings[$key])) {
+            $output[] = [
+                $value,
+                $target_strings[$key],
+            ];
+        }
```

Comment:
I am testing and it does have an impact on perf (for French, requests that were taking 0.17s are now taling 0.2s), I think it is because creating a multidimensionnal array with associative keys is more expensive than one without. That said, I can put inline comments to mention which one is the source and which one is the target string